### PR TITLE
[tests-only] Feature to debug specific test suites in CI

### PR DESCRIPTION
**Pull request by saw-jan**
## Description
adds feature to be able to run specific test suites in CI

## Related Issue
part of https://github.com/owncloud/QA/issues/670

## How Has This Been Tested?
- test environment:

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: .drone.star [hunk 1]
```diff
@@ -129,6 +133,8 @@ config = {
 				'apiWebdavUpload1',
 				'apiWebdavUpload2',
 			],
+			'skip': False,
+			'debugSuites': []
```
Comment (kiranparajuli589) [inline .drone.star:137]: imo default values needn't be used in `config`

Comment (saw-jan) [reply .drone.star:137]: Yeah, I place those so that anyone would know where to skip and debug suites

Comment (phil-davis) [reply .drone.star]: We do want the skip false in the config section - that makes it easy for people to change to skip true. I guess we will learn how to insert the other special debugSuites and skipExceptParts.

File: .drone.star [hunk 2]
```diff
@@ -1370,6 +1372,11 @@ def acceptance(ctx):
 		else:
 			suites = matrix['suites']
 
+		if 'debugSuites' in matrix and len(matrix['debugSuites']) != 0:
+			suites = {}
+			for suite in matrix['debugSuites']:
+				suites[suite] = suite
```
Comment (phil-davis) [inline .drone.star:1378]:
```suggestion
			if type(matrix['suites']) == "list":
				suites = {}
				for suite in matrix['debugSuites']:
					suites[suite] = suite
			else:
				suites = matrix['debugSuites']
```

There are 2 ways to specify a list of suites:
- as a dict key-value list - the keys are the suite names, and the values are the shortened names to use in the pipeline name
- as a list (ordinary array) - the items are the suite names, that are also used directly in the pipeline names

IMO the code here needs to handle both.

File: .drone.star [hunk 3]
```diff
@@ -1370,6 +1372,14 @@ def acceptance(ctx):
 		else:
 			suites = matrix['suites']
 
+		if 'debugSuites' in matrix and len(matrix['debugSuites']) != 0:
+			if type(matrix['suites']) == "list":
```
Comment (phil-davis) [inline .drone.star:1376]:
```suggestion
			if type(matrix['debugSuites']) == "list":
```

I guess we had better not assume that `suites` and `debugSuites` will always have the same type.

**Review by phil-davis — APPROVED** [review]

**Pull request merged**